src/tests/fs_tests.rs

```diff
@@ -2218,4 +2220,34 @@ pub mod fs_tests {
         assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
         lindrustfinalize();
     }
+    use crate::interface::IovecStruct;
+    #[test]
+    fn test_writev_syscall_stream() {
+        let _thelock = setup::lock_and_init();
+        let cage = interface::cagetable_getref(1);
+    
+        let fd = 1; // stdout
+    
+        // Prepare the iovec structures
+        let hello = b"Hello, ";
+        let world = b"world!";
+        let iovecs = [
+            IovecStruct {
+                iov_base: hello.as_ptr() as *mut c_void,
+                iov_len: hello.len(),
+            },
+            IovecStruct {
+                iov_base: world.as_ptr() as *mut c_void,
+                iov_len: world.len(),
+            },
+        ];
+    
+        // Call writev_syscall
+        let bytes_written = cage.writev_syscall(fd, iovecs.as_ptr(), iovecs.len() as i32);
+        
+        // Validate the results
+        assert_eq!(bytes_written, 13, "Bytes written do not match expected value");
```

Comment:
i think it would be good to check that the data from the buffers is written to the specified file and not some other file (instead of just checking the number of bytes written)